src/backend/partaj/core/api/referral.py

```diff
@@ -96,41 +98,96 @@ def get_permissions(self):
+        Create a empty referral as the client issues a POST on the referrals endpoint.
         """
-        form = ReferralForm(
-            {
-                # Add the currently logged in user to the Referral object we're building
-                "users": [request.user.id],
-                **{key: value for key, value in request.POST.items()},
-            },
-            request.FILES,
-        )
+        referral = models.Referral.objects.create()
+        referral.users.set([request.user])
+        referral.save()
+        return Response(data=ReferralSerializer(referral).data)
 
-        if form.is_valid():
+    def update(self, request, *args, **kwargs):
+        """
+        Update an existing referral.
+        """
+        referral = self.get_object()
+
+        referral.context = request.data.get("context")
+        referral.context = request.data.get("question")
+        referral.object = request.data.get("object")
+        referral.prior_work = request.data.get("prior_work")
+        referral.urgency_explanation = request.data.get("urgency_explanation")
+
+        if request.data.get("topic"):
+            try:
+                topic = models.Topic.objects.get(id=request.POST["topic"])
+                referral.topic = topic
+            except models.Topic.DoesNotExist:
+                return Response(
+                    status=400,
+                    data={
+                        "urgency_level": [
+                            f"{request.data.get('Topic')} is not a valid topic."
+                        ]
+                    },
+                )
+
+        referral.users.clear()
+        referral.users.set(request.data.getlist("users"))
+
+        if request.data.get("urgency_level"):
             # Do not create the referral until we can completely validate it: we need to first
             # make sure the urgency ID we received matches an existing urgency level.
             try:
                 referral_urgency = models.ReferralUrgency.objects.get(
-                    id=request.POST["urgency_level"]
+                    id=request.data.get("urgency_level")
                 )
             except models.ReferralUrgency.DoesNotExist:
                 return Response(
                     status=400,
                     data={
                         "urgency_level": [
-                            f"{request.POST['urgency_level']} is not a valid referral urgency id."
+                            f"{request.data.get('urgency_level')} is not a valid referral urgency id."
                         ]
                     },
                 )
-
-            # Create the referral from existing data
-            referral = form.save()
-
             # Add in the urgency level we found and the relevant unit
             referral.urgency_level = referral_urgency
+
+        referral.save()
+
+        # Create Attachment instances for the related files
+        files = request.FILES.getlist("files")
```

Comment:
I think we can do this like for referral answers: use separate routes for attachments. This way we live upload them as the user adds them to the widget.